core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -0,0 +1,424 @@
+import alluxio.util.executor.ExecutorServiceFactory;
+import alluxio.util.network.NetworkAddressUtils;
+import alluxio.wire.Address;
+import alluxio.wire.ConfigProperty;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterators;
+import org.apache.thrift.TProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.time.Clock;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * The default meta master.
+ */
+@NotThreadSafe
+public final class DefaultMetaMaster extends AbstractMaster implements MetaMaster {
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultMetaMaster.class);
+  private static final Set<Class<? extends Server>> DEPS =
+      ImmutableSet.<Class<? extends Server>>of(BlockMaster.class);
+
+  // Master metadata management.
+  private static final IndexDefinition<MasterInfo> ID_INDEX =
+      new IndexDefinition<MasterInfo>(true) {
+        @Override
+        public Object getFieldValue(MasterInfo o) {
+          return o.getId();
+        }
+      };
+
+  private static final IndexDefinition<MasterInfo> ADDRESS_INDEX =
+      new IndexDefinition<MasterInfo>(true) {
+        @Override
+        public Object getFieldValue(MasterInfo o) {
+          return o.getAddress();
+        }
+      };
+
+  /** Handle to the block master. */
+  private final BlockMaster mBlockMaster;
+
+  /** The clock to use for determining the time. */
+  private final Clock mClock = new SystemClock();
+
+  /** The master configuration store. */
+  private final ServerConfigurationStore mMasterConfigStore = new ServerConfigurationStore();
+  /** The worker configuration store. */
+  private final ServerConfigurationStore mWorkerConfigStore = new ServerConfigurationStore();
+  /** The server-side configuration checker. */
+  private final ServerConfigurationChecker mConfigChecker =
+      new ServerConfigurationChecker(mMasterConfigStore, mWorkerConfigStore);
+
+  /** Keeps track of standby masters which are in communication with the leader master. */
+  private final IndexedSet<MasterInfo> mMasters =
+      new IndexedSet<>(ID_INDEX, ADDRESS_INDEX);
+  /** Keeps track of standby masters which are no longer in communication with the leader master. */
+  private final IndexedSet<MasterInfo> mLostMasters =
+      new IndexedSet<>(ID_INDEX, ADDRESS_INDEX);
+
+  /** The address of this master. */
+  private final Address mMasterAddress = new Address()
+      .setHost(Configuration.get(PropertyKey.MASTER_HOSTNAME))
+      .setRpcPort(Configuration.getInt(PropertyKey.MASTER_RPC_PORT));
```

Comment:
This creates a hidden dependency that we update the configured `MASTER_RPC_PORT` in `AlluxioMasterProcess` before constructing this class. Consider passing in the RPC port through `MasterContext` instead